examples_utils/notebook_logging/gc_logger.py

```diff
@@ -160,10 +161,20 @@ def __new__(cls, ip):
                 # Prepare shared dict and populate with Nulls in schema format
                 cls._PAYLOAD.update(cls._COLUMN_TYPES)
 
-                # Create a short unique user ID
-                cls._UNIQUE_HASH = base64.urlsafe_b64encode(
-                    hashlib.md5(cls._CREATION_TIME.strftime("%Y-%m-%d %H:%M:%S.%f").encode("utf-8")).digest()
-                ).decode("ascii")[:12]
+                # Find existing user ID, or create one
+                userid_file = Path("/storage/.generated_user_id").resolve()
```

Comment:
Consider making it explicit that this is an IPU/graphcore created file

```suggestion
                userid_file = Path("/storage/.graphcore/generated_user_id").resolve()
                userid_file.parent.mkdir(parents=True, exists_ok=True)
```

Reply:
Cool done, thanks!